### PR TITLE
Ignore bogus "not installed" exception from PackageVersions

### LINKED ISSUE #1148: Infection no longer works via a composer global install?
Looks like maybe infection no longer works via a composer global install? https://app.circleci.com/pipelines/github/bdsl/psalm/218/workflows/5dfa1e19-20de-492c-9a58-95c9a03f61f9/jobs/934

```
#!/bin/bash -eo pipefail
php -d memory_limit=4G ~/.composer/vendor/bin/infection --coverage=build/phpunit --only-covered --threads=2 || echo 'Temporarily ignoring unexplained infection failure'


Fatal error: Uncaught OutOfBoundsException: Required package "infection/infection" is not installed: check your ./vendor/composer/installed.json and/or ./composer.lock files in /home/docker/project/vendor/ocramius/package-versions/src/PackageVersions/Versions.php:89
Stack trace:
#0 /home/docker/.composer/vendor/infection/infection/src/Console/Application.php(81): PackageVersions\Versions::getVersion('infection/infec...')
#1 /home/docker/.composer/vendor/infection/infection/bin/infection(83): Infection\Console\Application->__construct(Object(Infection\Container))
#2 {main}
  thrown in /home/docker/project/vendor/ocramius/package-versions/src/PackageVersions/Versions.php on line 89
Temporarily ignoring unexplained infection failure

CircleCI received exit code 0
```

_Originally posted by @bdsl in https://github.com/infection/infection/issues/705#issuecomment-597313370_

---

Possibly related to https://github.com/infection/infection/issues/876

### PR DESCRIPTION
**Pull request by sanmai**
Fixes #1148

### CODE REVIEW

File: src/Console/Application.php [hunk 1]
```diff
@@ -78,7 +82,20 @@ final class Application extends BaseApplication
...
+            // @codeCoverageIgnoreStart
+        } catch (OutOfBoundsException $e) {
+            if (preg_match('/package .*' . self::PACKAGE_NAME . '.* not installed/', $e->getMessage()) === 0) {
+                throw $e;
+            }
+
+            // We have a bogus exception: how can Infection be not installed if we're here?
+            $version = 'not-installed';
+        }
+        // @codeCoverageIgnoreEnd
```
Comment (sanmai) [inline src/Console/Application.php]: This happens in very peculiar circumstances which, I think, just not worth the time to be reproduced. Therefore not covered by default.

**Comment by sanmai** [comment]
Thank you @maks-rafalko

File: src/Console/Application.php (hunk 1 above)
Comment (theofidry) [reply src/Console/Application.php]: So what version will be displayed there? Wasn't a case also when you were globally installing infection with `--no-scripts`?

Comment (sanmai) [reply src/Console/Application.php]: With `--no-scripts` there's no exception here, IIRC. That issue is still open.

**Pull request merged**